arm/ui/settings/settings.py

```diff
@@ -176,27 +188,43 @@ def save_settings():
     Page - save_settings
     Method - POST
     Overview - Save arm ripper settings from post. Not a user page
+    This function writes to arm.yaml
     """
     # Load up the comments.json, so we can comment the arm.yaml
+    app.logger.info("Saving ARM Ripper settings")
     comments = ui_utils.generate_comments()
     success = False
     arm_cfg = {}
+    form_name = "arm ripper settings"
+    app.logger.debug("Generating a temporary instance of SettingsForm")
     form = SettingsForm()
+    app.logger.info(f"Form errors are: {form.errors}")
     if form.validate_on_submit():
+        app.logger.debug("Saving Ripper settings")
         # Build the new arm.yaml with updated values from the user
-        arm_cfg = ui_utils.build_arm_cfg(request.form.to_dict(), comments)
-        # Save updated arm.yaml
-        with open(cfg.arm_config_path, "w") as settings_file:
-            settings_file.write(arm_cfg)
-            settings_file.close()
-        success = True
+        arm_cfg = ui_utils.build_arm_cfg(form_data=request.form.to_dict(), comments=comments)
+        app.logger.debug(f"Cleansed Form Data: \r\n{arm_cfg}")
+        try:
+            # Save updated arm.yaml
+            app.logger.debug(f"routes.save_settings: Saving new arm.yaml: {cfg.arm_config_path}")
+            with open(cfg.arm_config_path, "w") as settings_file:
+                settings_file.write(arm_cfg)
+            success = True
+        except Exception as e:
+            app.logger.exception(f"Error saving arm.yaml: {e}")
+            flash(f"Error saving arm.yaml: {e}", "error")
```

Comment:
I thought we moved away from flash() ? And were using toasts/notification system ?

Because the flash() doesn't update till the next page load, so there will be no user feedback until they go to another page. You can see this, by failing to update the settings and then changing the page, a badly formed box will appear. But only after then user has already changed the page.

Reply:
I'm not sure if we moved away from it in the past.
I was looking into handling the 'Flash' via ajax.

Your correct throwing the error to the user on form submit is best, waiting for page refresh is ridiculous.
But until I can sort out the ajax end, it's best to leave these in as a sort of 'placeholder', IMHO.